service_plans/v1/service_plans_suite_test.go

```diff
@@ -0,0 +1,101 @@
+	"time"
+
+	"github.com/cloudfoundry-incubator/cf-test-helpers/workflowhelpers"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/cloudfoundry-incubator/cf-performance-tests/helpers"
+)
+
+var testConfig = helpers.NewConfig()
+var prefix = testConfig.GetNamePrefix()
+var testSetup *workflowhelpers.ReproducibleTestSuiteSetup
+var ccdb *sql.DB
+var uaadb *sql.DB
+var ctx context.Context
+
+const (
+	orgs = 10000
+	serviceOfferings = 300
+	servicePlansPublic = 10
+	servicePlansPrivateWithoutOrgs = 10
+	servicePlansPrivateWithOrgs = 10
+	orgsPerLimitedServicePlan = 200
+)
+
+var _ = BeforeSuite(func() {
+
+	testSetup = workflowhelpers.NewTestSuiteSetup(&testConfig)
+	testSetup.Setup()
+	ccdb, uaadb, ctx = helpers.OpenDbConnections(testConfig)
+
+	fmt.Printf("%v Starting to seed database with testdata...\n", time.Now().Format(time.RFC850))
+
+	helpers.ImportStoredProcedures(ccdb, ctx)
+
+	serviceBrokerId := createServiceBroker()
+
+	createOrgStatement := fmt.Sprintf("SELECT FROM create_orgs(%d)", orgs)
+	helpers.ExecuteStatement(ccdb, ctx, createOrgStatement)
+
+	createPublicServicePlansStatement := fmt.Sprintf("SELECT FROM create_services_and_plans(%v, %v, %v, %v, %v)",
+		serviceOfferings, serviceBrokerId, servicePlansPublic, true, 0)
+	helpers.ExecuteStatement(ccdb, ctx, createPublicServicePlansStatement)
+
+	createPrivateServicePlansStatement := fmt.Sprintf("SELECT FROM create_services_and_plans(%v, %v, %v, %v, %v)",
+		serviceOfferings, serviceBrokerId, servicePlansPrivateWithoutOrgs, false, 0)
+	helpers.ExecuteStatement(ccdb, ctx, createPrivateServicePlansStatement)
+
+	createPrivateServicePlansWithOrgsStatement := fmt.Sprintf("SELECT FROM create_services_and_plans(%v, %v, %v, %v, %v)",
+		serviceOfferings, serviceBrokerId, servicePlansPrivateWithOrgs, false, orgsPerLimitedServicePlan)
+	helpers.ExecuteStatement(ccdb, ctx, createPrivateServicePlansWithOrgsStatement)
+
+	regularUserGUID := helpers.GetUserGUID(testSetup.RegularUserContext(), testConfig)
+	orgsAssignedToRegularUser := orgs/2
+	assignUserAsOrgManager := fmt.Sprintf("SELECT FROM assign_user_as_org_manager('%s', %d)", regularUserGUID, orgsAssignedToRegularUser)
+	helpers.ExecuteStatement(ccdb, ctx, assignUserAsOrgManager)
+
+	fmt.Printf("%v Finished seeding database.\n", time.Now().Format(time.RFC850))
+})
+
+var _ = AfterSuite(func() {
+	fmt.Printf("%v Starting cleanup testdata...\n", time.Now().Format(time.RFC850))
+	helpers.CleanupTestData(ccdb, uaadb, ctx, testConfig)
+	fmt.Printf("%v Finished cleanup testdata...\n", time.Now().Format(time.RFC850))
+	err := ccdb.Close()
+	if err != nil {
+		log.Print(err)
+	}
+
+	if uaadb != nil {
+		err = uaadb.Close()
+		if err != nil {
+			log.Print(err)
+		}
+	}
+})
+
+func TestServicePlans(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecsWithDefaultAndCustomReporters(t, "ServicePlansTest Suite", []Reporter{helpers.ConfigureJsonReporter(t, &testConfig, "service-plans")})
+}
+
+func createServiceBroker() int {
```

Comment:
Is the broker id really needed? For the service keys tests, I've created services without such an id.

Reply:
We think that because of the service table definition `Foreign-key constraints:
    "fk_services_service_broker_id" FOREIGN KEY (service_broker_id) REFERENCES service_brokers(id)` we need the service_broker_id

Reply:
You are right. It's needed when showing the service plans as it's loading the broker eagerly. But during the creation it's actually not needed...

Reply:
You are right. It's needed when showing the service plans as it's loading the broker eagerly. But during the creation it's actually not needed...